src/test/ui/invalid_const_promotion.rs

```diff
@@ -22,7 +22,7 @@ use std::process::{Command, Stdio};
 const fn bar() -> usize { 0 - 1 }
 
 fn foo() {
-    let _: &'static _ = &bar();
+    let _: &'static _ = &bar();  // FIXME: No longer compiles
```

Comment:
This test no longer compiles; the error is:

    error[E0080]: evaluation of constant value failed
      --> C:\Users\david\dev\rust\src/test\ui\invalid_const_promotion.rs:22:27
       |
    LL | const fn bar() -> usize { 0 - 1 }
       |                           ^^^^^ attempt to subtract with overflow
    
    error[E0080]: evaluation of constant expression failed
      --> C:\Users\david\dev\rust\src/test\ui\invalid_const_promotion.rs:25:25
       |
    LL |     let _: &'static _ = &bar();
       |                         ^-----
       |                          |
       |                          referenced constant has errors

The test name `invalid_const_promotion` suggests that we don't want `bar()` to be evaluated here at compile time. Is that correct? If so, then this is indeed a bug introduced by my implementation of this PR.

Reply:
This is a test for an unstable internal feature(rustc_promotable). Just give bar an argument to get the old behaviour back

Reply:
Ah - ok, thanks!